### PR TITLE
Feature/updates for gdas app

### LINKED ISSUE #1043: build_all.sh fails when gdas.cd is present
**Expected behavior**
`./build_all.sh` on Orion from develop @ [833b8f4](https://github.com/NOAA-EMC/global-workflow/commit/833b8f4c124fb48b5a04bc0f82c2b5fc9df1165c) should run to completion when `sorc/gdas.cd` is present.

**Current behavior**
`./build_all.sh` fails on Orion when it gets to gdas.cd.
```
Orion-login-3[23] rtreadon$ ./build_all.sh -v
INFO: settings in config file: gfs_build.cfg
Building ufs_model (ufs_model) ........................ yes
Building ww3_prepost (ww3_prepost) .................... yes
Building gsi_enkf (gsi_enkf) .......................... yes
Building gsi_utils (gsi_utils) ........................ yes
Building gsi_monitor (gsi_monitor) .................... yes
Building gldas (gldas) ................................ yes
Building UPP (upp) .................................... yes
Building ufs_utils (ufs_utils) ........................ yes
Building gfs_wafs (gfs_wafs) .......................... yes
Building gfs_utils (gfs_utils) ........................ yes
INFO: partial build settings:
  Build_ufs_model: True
  Build_ww3_prepost: True
  Build_gsi_enkf: True
  Build_gsi_utils: True
  Build_gsi_monitor: True
  Build_ww3_prepost: True
  Build_reg2grb2: True
  Build_gldas: True
  Build_upp: True
  Build_ufs_utils: True
  Build_gfs_wafs: True
  Build_gfs_utils: True
 .... Building WW3 pre and post execs ....
 .... Building forecast model ....
 .... Building gsi and enkf ....
 .... Building gsi utilities ....
./build_all.sh: line 170: Build_gdas: parameter null or not set
Orion-login-3[24] rtreadon$
```

**Machines affected**
Only tested on Orion.

**To Reproduce**
1.  clone g-w `develop` at [833b8f4](https://github.com/NOAA-EMC/global-workflow/commit/833b8f4c124fb48b5a04bc0f82c2b5fc9df1165c)
2.  `cd sorc`
3.  I did `checkout.sh -g -u` to clone both DA engines for comparative studies, but `checkout.sh -u` is sufficient to see the failure.
4. `./build_all.sh`

**Possible Implementation**
As a test, 
1. add `Build_gdas` to `Build_prg` list in `partial_build.sh`
2. add `Building gdas (gdas) .................................. yes` to `gfs_build.cfg`
3. `./build_all.sh` runs to completion

Since `./checkout.sh` supports both GSI (`-g`) and UFS (`-u`) DA builds, should `partial_build.sh` and `gfs_build.cfg` include both _gsi_enkf_ and _gdas_?

### LINKED ISSUE #1067: Update g-w to cycle GDASApp
**Expected behavior**
g-w develop `jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP` should run to completion.

**Current behavior**
`JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP` fails with the traceback

```
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
+++ [82]: env
+++ [82]: grep -o '[^=]*CONDA[^=]*'
+++ [82]: grep -v 'CONDA_ENVS_PATH\|CONDA_PKGS_DIRS\|CONDARC'
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
++ [82]: unset CONDA_SHLVL CONDA_EXE _CE_CONDA CONDA_PYTHON_EXE
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
++ [82]: unset prefix
/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 82: BASH_SOURCE: unbound variable
++ [82]: conda deactivate

CommandNotFoundError: Your shell has not been properly configured to use 'conda deactivate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.


/work/noaa/da/Russ.Treadon/git/global_workflow/develop/jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP: line 1: BASH_SOURCE: unbound variable
+ [1]: postamble JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP 1665512197 1
+ preamble.sh[68]: set +x
End JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP at 18:16:49 with error code 1 (time elapsed: 00:00:12)
```

**Machines affected**
Running C96L127 parallel on Orion using GDASApp based (UFS-based) DA

**To Reproduce**
1. Install g-w develop at e8ef5fc6
2. set up EXPDIR with `export DO_JEDIVAR="YES"` in `config.base`
3. populate ROTDIR for UFS-based DA
4. submit gdasatmanalprep

A log file with the reported error is `/work/noaa/stmp/rtreadon/comrot/prgdasens4/logs/2021122100/gdasatmanalprep.log`

**Context**
Closed issue #1015 reported another case of `$BASH_SOURCE` errors.   #1015 differs from the failure reported in this issue.

**Detailed Description**
The section of `JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP` in which the failure occurs is 

```
# NOTE BELOW IS A HACK FOR TESTING
# PLEASE FIX THIS LATER
# ASK @aerorahul
# HOW TO HANDLE DIFFERENT COMPILERS/ETC. FOR MODEL VS DA
# PROD_UTIL, ETC. DO NOT EXIST FOR JEDI MODULE VERSIONS
module purge
module use $HOMEgfs/sorc/gdas.cd/modulefiles
module load GDAS/orion
export PYTHONPATH=$HOMEgfs/sorc/gdas.cd/ush/:$PYTHONPATH
```
According to gdasatmanalprep.log the failure occurs on the `module purge` line.

**Possible Implementation**
As noted in the comments in `JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP`, the sequence of `module purge`, `use`, and `load` should not be in the j-job.    This section of the job needs to be refactored.   What's does the g-w team recommend?

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
**Description**
Updates the GDASapp version to a newer version that builds correctly. The former version no longer builds because submodules point to develop branches that are no longer compatible.

Moves module loads out of the j-jobs and into their appropriate place in the rocoto job. A new load module script is added to handle the different module set needed for UFSDA.

Also temporarily turns off strict mode for the UFSDA jobs to avoid PS1 unbound error in conda.

Fixes #1043 
Fixes #1067

**Type of change**
- [x] Bug fix - restore ability to build and cycle with GDASApp

**How Has This Been Tested?**
The changes in this PR along with changes in GDASApp PR [#162](https://github.com/NOAA-EMC/GDASApp/pull/162) have been tested in a low resolution parallel on Hera.

- [x] Clone and Build tests on Hera and Orion.   _(note:  it is not yet possible to build GDASApp on WCOSS2)_
- [x] Cycled test on Hera.
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

### CODE REVIEW

File: jobs/rocoto/atmanalpost.sh [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #! /usr/bin/env bash
 
+export STRICT="NO"
```
Comment (WalterKolczynski-NOAA) [inline jobs/rocoto/atmanalpost.sh:3]: Why are you turning strict off in all these? Especially to just turn it back on immediately after (though that doesn't really do anything unless you call `set_strict` later).

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: jobs/rocoto/atmanalpost.sh (hunk 1 above)
Comment (RussTreadon-NOAA) [reply jobs/rocoto/atmanalpost.sh:3]: I neither like nor advocate these lines.   They are, for the time being, necessary due to the problems documented in issue #1067.   If @aerorahul or you have a better solution which allows us to cycle UFS-DA on Orion, please let us know.

Comment (WalterKolczynski-NOAA) [reply jobs/rocoto/atmanalpost.sh:3]: Okay, I don't want to hold up this PR. But, if you can provide instructions on how to set up an experiment with UFSDA, I'll see if I can figure out what's going on.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
`Externals.cfg` needs the same hash updates as `checkout.sh`, then I think this is ready to go.

File: jobs/rocoto/atmanalpost.sh (hunk 1 above)
Comment (CoryMartin-NOAA) [reply jobs/rocoto/atmanalpost.sh:3]: It's related (99% sure) to module purge/module load with the conda python env we are using. It is not a good/permanent solution. @WalterKolczynski-NOAA do you have a workflow python env that works properly that we can install needed utilities in?

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**